include/envoy/http/filter.h

```diff
@@ -222,6 +222,19 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
    */
   virtual void addDecodedData(Buffer::Instance& data, bool streaming_filter) PURE;
 
+  /**
+   * Decode data directly to subsequent filters in the filter chain. This method is used in
+   * advanced cases in which a filter needs full control over how subsequent filters view data,
+   * and does not want to make use of HTTP connection manager buffering. Using this method allows
```

Comment:
"and does not want to make use of HTTP connection manager buffering"

Consider making this more explicit, how this filter has an implicit contract to not StopIterationAndBuffer etc. from decodeData calls.    It's potentially confusing if the next filter is say the buffering filter and the resultant data will land directly in the HCM buffer

Also either here or above it might be worth calling out explicit differences between decodeData and addDecodedData, since this does add decoded data, it just adds it further down the chain.  bonus points if you can think of a more descriptive name, since we already have a number of decodeData calls.   Ideally something more terse than addDecodedDataToUpstreamFilters but hopefully comparably descriptive.

Reply:
OK let me think about this and I will update.